src/components/TodoList.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <ul class="list-group">
+    <div v-for="todo in todos" :key="todo.id" class="mb-1">
+      <TodoItem
+        v-if="!isTodoBeingEdited(todo)"
+        :todo="todo"
+        :handle-toggle-completed="handleToggleCompleted"
+        :handle-click-edit="handleClickEdit"
+        :handle-click-remove="handleClickRemove"
+      />
+      <EditTodoForm
+        v-else
+        :todo="todo"
+        :handle-submit-edited-todo-and-reset="handleSubmitEditedTodoAndReset"
+        :handle-cancel-edit="handleCancelEdit"
+      />
+    </div>
+  </ul>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType, reactive } from "vue";
+import Todo from "@/models/todo";
+import TodoItem from "@/components/TodoItem.vue";
+import EditTodoForm from "@/components/EditTodoForm.vue";
+
+import type {
+  HandleToggleCompleted,
+  HandleClickEdit,
+  HandleClickRemove,
+} from "@/components/TodoItem.vue";
+import type {
+  HandleSubmitEditedTodoAndReset,
+  HandleCancelEdit,
+} from "@/components/EditTodoForm.vue";
+
+export type HandleSubmitEditedTodo = (editedTodo: Todo) => void;
+
+type State = {
+  todoIdBeingEdited: string | null;
```

Comment:
好み

tsでの書きやすさ考えると、こういうのはnullではなくundefined使う派です。

```suggestion
  todoIdBeingEdited?: string;
```

Reply:
なるほど！

修正しました。